lib/datadog/tracing/component.rb

```diff
@@ -29,6 +29,9 @@ def build_tracer(settings, agent_settings)
 
         subscribe_to_writer_events!(writer, sampler, settings.tracing.test_mode.enabled)
 
+        current_schema_version = SpanAttributeSchema
+                                   .get_schema_version_numeric(Datadog.configuration.tracing.span_attribute_schema)
```

Comment:
It's not safe to call `Datadog.configuration` inside this method, as the configuration itself is being modified when `Component`s are being build.

To address that, this method receives the configuration in the `settings` argument:
```suggestion
                                   .get_schema_version_numeric(settings.tracing.span_attribute_schema)
```